### ISSUE TITLE
class_delegate_protocol warning triggered by the Actor protocol

### ISSUE THREAD

**Issue by bagofstars** [realm/SwiftLint#6054]
### New Issue Checklist

- ✅ I've Updated SwiftLint to the latest version.
- ✅ I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

`class_delegate_protocol` doesn't recognize the Actor protocol as suitable for being weakly referenced.

```swift
protocol FooDelegate: Actor { }

actor Bar {
    weak var delegate: (any FooDelegate)?	// raises warning: Class Delegate Protocol Violation: Delegate protocols should be class-only so they can be weakly referenced (class_delegate_protocol)
}
```

Since Actor protocol [already conforms to AnyObject](https://developer.apple.com/documentation/swift/actor) I believe it can also be whitelisted.

Bonus idea for further consideration: how about renaming the rule into `reference_type_delegate_protocol`? (I don't have a strong opinion about this one.)

### Environment

* SwiftLint version: 0.59.1
* Xcode version: 16.3
* Installation method used: Homebrew

**Comment by carloshpdoc** [comment]
Hi! I'd like to work on this issue.  It's ok?

**Comment by mildm8nnered** [comment]
> Hi! I'd like to work on this issue. It's ok?

Absolutely!

**Comment by wvteijlingen-npo** [comment]
Has there been any progress on this issue? (@carloshpdoc)

**Comment by SimplyDanny** [comment]
> Has there been any progress on this issue? ([@carloshpdoc](https://github.com/carloshpdoc))

The issue has been fixed by #6067.
